### PR TITLE
Do not allow selecting text in the Paginator component of the Product editor

### LINKED ISSUE #40501: [FIX] When clicking fast on the pagination, it selects text around it
## Description

When clicking fast on the pagination, it selects text around it

https://github.com/woocommerce/woocommerce/assets/13437655/bc10254d-5def-428a-8e60-5b2f4131af2d

This shouldn't happen

## Acceptance criteria

- [ ] When navigating fast between pages on variations, text close to the buttons should not be selected

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/40501

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure to enable the New Product Feature ( WooCommerce -> Advances)
2. Before checking this branch (trunk)
  a. Go to the Product Editor -> Variations
  b. Ensure to add some variations
  c. Confirm if you click again and again in the variations paginator, it selects text (#40501)
3. Applying this PR changes
  a. Go to the Product Editor -> Variations
  b. Confirm it isn't possible to select text when clicking on the paginator

https://github.com/woocommerce/woocommerce/assets/77539/55351b02-4556-4e59-805f-e5a8ddbd31fe

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [x] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/changelog/update-do-not-allow-selecting-text-in-paginator [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: enhancement
+
+Do not allow to select text in the Paginator component
```
Comment (octaedro) [inline packages/js/product-editor/changelog/update-do-not-allow-selecting-text-in-paginator]: I think that this file should live under `packages/js/components/changelog`. This is why we see [this error here](https://github.com/woocommerce/woocommerce/actions/runs/6853041724/job/18632956482?pr=41405).

**Review by octaedro — APPROVED** [review]
Good job @retrofox! This is testing well here, I only left a small comment related to the changelog file. After addressing the change, this PR will be ready to merge.
I'll pre-approve this.

File: packages/js/product-editor/changelog/update-do-not-allow-selecting-text-in-paginator (hunk 1 above)
Comment (retrofox) [reply packages/js/product-editor/changelog/update-do-not-allow-selecting-text-in-paginator]: Good! Updating cc @mattsherman

**Comment by retrofox** [comment]
> After addressing the change, this PR will be ready to merge. I'll pre-approve this.

✔️

**Pull request merged**